machine-learning/05-machine_learning_exercise.Rmd

```diff
@@ -0,0 +1,245 @@
+that was processed with refine.bio. 
+
+Here's an excerpt of the description from the authors:
+
+> To identify patterns of drug-induced gene modulation that occur across different cell types, we measured gene expression changes across NCI-60 cell lines after exposure to 15 anticancer agents.
+
+**We've supplied the following to you:**
+
+* The gene expression matrix (`data/GSE116436/GSE116436_filtered.tsv`)
+* The sample metadata (`data/GSE116436/metadata_GSE116436_cleaned.tsv`)
+* Sample-sample Pearson correlation matrix (`data/GSE116436/GSE116436_sample_correlation.RDS`)
+* PCA (`prcomp` output) (`models/NCI60_prcomp_results.RDS`)
+* A PLIER model (`models/NCI60_PLIER_model.RDS`)
+
+_Note that we've only made a subset of samples available and we've already
+cleaned the metadata for you.
+If you want to learn more about how the metadata was cleaned, ask an
+instructor._
+
+We're going to look specifically at **pathways that change with treatment**, 
+but we precomputed the other objects in case you would like to understand what 
+samples tend to group together or what is driving the variation in the data.
+
+We'll use `dplyr` and `ggplot2` quite a bit in this notebook, so use the chunk
+below to load them.
+
+```{r}
+
+```
+
+The metadata is available at `data/data/GSE116436/metadata_GSE116436_cleaned.tsv`.
+Read it in with `readr` and save it as `metadata_df`
+
+```{r}
+metadata_df <- 
+```
+
+What metadata do we have for these samples ? 
+Use this chunk to explore `metadata_df`.
+
+```{r}
+
+```
+
+## What pathways are perturbed by the drugs in this experiment?
+
+The PLIER model we've precomputed is available `models/NCI60_PLIER_model.RDS`.
+Read it in with `readr` and save it as `plier_results`.
+
+```{r}
+plier_results <-
+```
+
+The `summary` element of what is returned by PLIER can tell us about
+associations, you can access it with `$` and it is a `data.frame`.
+
+Use this chunk to view the PLIER model `summary`.
+
+```{r}
+
+```
+
+Save the summary information as it's own data.frame called `summary_df`. 
+We'll use this object to find out more about our model. 
+
+```{r}
+summary_df <-
+```
+
+The latent variables from this model will be associated with a pathway if the
+gene expression data supports that there is some variation in the expression
+levels of genes in that pathway.
+It could be due to treatment with a drug or variation could arise from tissue- 
+or cell line-specific differences.
+
+### Is there a latent variable associated with the proteasome pathway?
+
+We can use a chain of `dplyr` functions to filter `summary_df`.
+Let's call `FDR < 0.05` associated; this is the same cutoff we used in the
+PLIER notebook earlier.
+
+```{r}
+summary_df %>%
+  dplyr::filter(,  # significant associations only
+                # grepl returns TRUE when the pattern "proteasome" is found
+                # in the pathway column
+                grepl("proteasome", pathway, ignore.case = TRUE))
```

Comment:
Might consider using `dplyr::contains()` for this since I don’t think we’ve introduced them to grep functions at all. This way we can stick to the tidyverse theme.

Reply:
Good call. I'm dropping `dplyr::` as well here.

Reply:
Dug into this a bit more -- I believe the preferred way to "keep it in the tidyverse" here is to use `stringr::str_detect` (see: [this SO post](https://stackoverflow.com/questions/22850026/filtering-row-which-contains-a-certain-string-using-dplyr)). But to ignore case we'd have to do `stringr::fixed("proteasome", ignore.case = TRUE))` within `str_select` and I think that becomes more confusing if you see it for the first time.

Reply:
Hmmm... that makes it harder to say. Your call.

Reply:
I'm going to stick with `grepl`.